### PR TITLE
Added rAssure_c_dp(ring r)

### PR DESCRIPTION
**Pull request by adipopescu**

### CODE REVIEW

File: kernel/combinatorics/hdegree.cc [hunk 1]
```diff
@@ -1030,6 +1030,10 @@ void scComputeHC(ideal S, ideal Q, int ak, poly &hEdge, ring tailRing)
     hComp(hexist, hNexist, k, hexist, &hNstc);
   else
     hNstc = hNexist;
+  if (hNexist==0)
```
Comment (hannes14) [inline kernel/combinatorics/hdegree.cc]: Why this? If hNexist would be 0, this routine must not be called anyway:
hEdge would be undefined!
Proposal: change to assume(hNexist>0);

File: libpolys/polys/kbuckets.cc [hunk 2]
```diff
@@ -13,7 +13,7 @@
 //#include <kernel/pShallowCopyDelete.h>
 #include <coeffs/coeffs.h>
 #include <polys/monomials/ring.h>
-//#include <kernel/p_Procs.h>
```
Comment (hannes14) [inline libpolys/polys/kbuckets.cc]: This has no effect and libpolys/polys/templates/p_Procs.h must not be used outside of
the template stuff!

File: libpolys/polys/monomials/p_polys.cc [hunk 3]
```diff
@@ -4529,8 +4529,8 @@ static inline unsigned long GetBitFields(const long e,
 // if (ev1 & ~ev2) then exp1 does not divide exp2
 unsigned long p_GetShortExpVector(const poly p, const ring r)
 {
-  assume(p != NULL);
```
Comment (hannes14) [inline libpolys/polys/monomials/p_polys.cc]: p_GetShortExpVector is one of the routine that must not be called with p==NULL.
Returning 0 is a reallay bad idea as 0 stand here for p being a NON-ZERO- constant polynomial.
If this routine is called with p==NULLm the caller is wrong!

File: libpolys/polys/monomials/p_polys.cc [hunk 4]
```diff
@@ -4529,8 +4529,8 @@ static inline unsigned long GetBitFields(const long e,
...
   if (p == NULL) return 0;
```
Comment (adipopescu) [inline libpolys/polys/monomials/p_polys.cc]: This line was added in this commit. I simply changed the order of assume and this line. 
f3094ae7 (Oleksandr Motsak  2012-07-23 17:29:18 +0200 4532)   if (p == NULL) return 0;

I will leave it as before my commit.

File: libpolys/polys/kbuckets.cc (hunk 2 above)
Comment (adipopescu) [inline libpolys/polys/kbuckets.cc]: Understood

File: kernel/combinatorics/hdegree.cc (hunk 1 above)
Comment (adipopescu) [inline kernel/combinatorics/hdegree.cc]: i will change it with your proposal.

**Comment by adipopescu** [comment]
These commits are already in #718

**Pull request closed without merging**